lib_eio_luv/eio_luv.ml

```diff
@@ -543,6 +543,44 @@ module Low_level = struct
       Handle.of_luv ~sw sock
   end
 
+  module Pipe = struct
+    type t = Luv.Pipe.t
+
+    let init ?for_handle_passing () = Luv.Pipe.init ~loop:(get_loop ()) ?for_handle_passing () |> or_raise
+
+    let to_luv x = x
+
+    let to_handle ?close_unix ~sw t =
+      Handle.of_luv ?close_unix ~sw t
+  end
+
+  module Process = struct
+    type t = Luv.Process.t * (int * int64) Promise.t
+
+    (* We have to wrap the redirections in order to enforce the use of
+       our low-level pipe so we can use [get_loop ()] in the init function
+       for the pipe. *)
+    type redirection = Luv.Process.redirection
+
+    let to_parent_pipe ?readable_in_child ?writable_in_child ?overlapped ~fd ~parent_pipe () =
+      let parent_pipe = Pipe.to_luv parent_pipe in
+      Luv.Process.to_parent_pipe ?readable_in_child ?writable_in_child ?overlapped ~fd ~parent_pipe ()
+
+    let inherit_fd = Luv.Process.inherit_fd
+
+    let await_exit (_, status) = Promise.await status
+
+    let stop (t, _) = Luv.Process.kill t Luv.Signal.sigkill |> or_raise
```

Comment:
this could probably be implemented on top of a more generic `val kill: t -> signal -> unit` that could allow sending a signal to the subprocess.

Reply:
Good point thanks! I suppose as a low-level API it's enough to just give people the power to send whatever signals they want. I added this under a `send_signal` function which I prefer as a name over `kill` :))